test/promise.js

```diff
@@ -152,4 +152,37 @@ describe('Promise', function () {
       }, 50);
     });
   });
-}); 
+  describe('Promise.prototype.then', function() {
+    var spy1 = sinon.spy();
+
+    function Subclass() {
+      spy1();
+      Promise.apply(this, arguments);
+    }
+
+    Object.setPrototypeOf(Subclass.prototype, Promise.prototype);
+
+    Subclass.prototype.then = function() {
+      return Promise.prototype.then.apply(this, arguments);
+    };
+
+    it('subclassed Promise resolves to subclass', function() {
+      var prom = new Subclass(function(resolve) {
+        resolve();
+      }).then(function() {
+        assert(spy1.called);
+        assert(prom instanceof Subclass);
+      });
+    });
+    it('subclassed Promise rejects to subclass', function() {
+      var spy2 = sinon.spy(),
+        prom = new Subclass(function(_, reject) {
+          reject();
+        }).then(spy2, function() {
+          assert(spy1.called);
```

Comment:
Same as above. Let's move the asserts outside the `.then` and use the more explicit `.calledTwice`